<launch>
    <arg name="model" />
    <arg name="gui" default="false" />

    <param name="use_sim_time" value="false"  />
    <param name="robot_description" textfile="$(find simulation_core)urdf/autolabor_mini.urdf" />
    <param name="use_gui" value="$(arg gui)" />

    <node pkg="simulation_core" type="simulation_core" name="simulation_core" output="screen">
    </node>

    <node name = "control_node" pkg="controller" type="controller">
        <param name="linear_min" value="0.2" />
        <param name="linear_max" value="2.0" />
        <param name="linear_step" value="0.2" />
        
        <param name="angular_min" value="0.5" />
        <param name="angular_max" value="3.0" />
        <param name="angular_step" value="0.5" />

        <param name="map_frame" value="odom"/>
        <param name="odom_frame" value="odom"/>
        <param name="base_link_frame" value="base_link"/>
        <param name="real_map_frame" value="real_map"/>

        <param name="noise_v_linear" value="0" />
        <param name="noise_v_theta" value="0" />
        
        <param name="max_a_linear" value="999.0" />
        <param name="max_a_theta" value="999.0" />

        <param name="max_v_linear" value="1.0" />
        <param name="max_v_linear" value="1.57"/>

        <param name="rate" value="100" />
    </node>

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

    <node name="rviz" pkg="rviz" type="rviz" />

    <!--node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0.0 0.0 0.0 0.0 0.0 0.0 /map /odom 100" /-->

</launch>